src/stores/store.ts

```diff
@@ -0,0 +1,91 @@
+import { EventEmitter } from 'events'
+import { Status, ModuleProfile, Api, PluginProfile, ApiEventEmitter } from '../types'
+import { defaultProfile } from '../profile'
+import { ApiFactory } from 'src/engine/api-factory';
+
+export interface State {
+  status: Status
+}
+
+/** Create an empty state */
+function createState<T extends State>(): T {
+  return {
+    status: {} as Status,
+  } as T
+}
+
+export const StoreProfile: Partial<ModuleProfile> = {
+  events: ['statusChanged'],
+  notifications: {}
```

Comment:
should this include 
```

 {
 'theme': ['switchTheme']
 }


```

Reply:
I we merge `Store` and `ApiFactory` yes. Right now it's in the `defaultProfile`...

Reply:
then we should probably merge them